rig/machine_control/regions.py

```diff
@@ -0,0 +1,200 @@
+"""SpiNNaker machine regions.
+
+.. todo::
+    Neaten this documentation!
+
+Regions (not to be confused with rig Regions) are used to specify areas of a
+SpiNNaker machine for the purposes of transmitting nearest neighbour packets or
+for determining which chips should be included in any flood-fill of data or
+application loading.
+
```

Comment:
The region specification system is defined in "Managing Big SpiNNaker Machines" By Steve Temple.

I've asked ST if there is URL for this doc and am awaiting a reply.